# **Контрольное задание на допуск к экзамену.**

**Вариант 3**


Cуществует сервис для аренды электросамокатов
«Самокаты».
Сервис позволяет воспользоваться самокатом от Юрента через приложение Т-Банка.
Чтобы арендовать самокат, клиенту необходимо зайти в приложение, открыть сервис
«Самокаты» и отсканировать QR-код на руле самоката. В этот момент у клиента
со счета списывается фиксированная сумма в качестве взноса. В конце поездки
также списывается итоговая сумма за поездку.
Данные охватывают сезон аренды самокатов 2024 года (с апреля по октябрь 2024).

**Описание полей:**

order_rk Идентификатор заказа (поездки)

party_rk_id Идентификатор клиента

minute_cost Стоимость минуты

hold_amount Размер суммы, которая замораживается на счете в момент
взятия самоката

transport_model Название модели самоката

created_dttm Дата и время создания заказа

distance_km Километраж поездки

book_start_dttm Дата и время начала поездки

book_end_dttm Дата и время завершения поездки

book_time_zone_cd Часовой пояс

local_book_start_dttm Дата и время начала поездки в часовом поясе человека, который брал самокат

loyalty_accrual_rub_amt Размер выплаченного кэшбэка в рублях

nominal_price_rub_amt Стоимость поездки

loyalty_accrual_bns_amt Размер выплаченного кэшбэка в бонусах (если оплата
происходила с помощью кредитной карты)

Gender_cd Пол клиента

lvn_state_nm Регион проживания человека

Age Возраст клиента

Education_level
Уровень образования клиента:
- SCH — начальное, среднее
- GRD — высшее
- UGR — неполное высшее
- PGR — два высших
- ACD — ученая степень

marital_status_cd Семейный статус человека:
- IMR — состоит в незарегистрированном браке
- MAR —женат/замужем
- DLW — не проживает с супругом(ой)
- OMR — состоит в зарегистрированном браке
- CIV — гражданский брак
- UNM — холост/не замужем
- DIV — разведен(а)
- FAL — никогда не состоял(а) в браке
- WID — вдовец/вдова

**Задание:**

1. Кто эти клиенты, которее чаще других использует сервис Самокаты? Целевая аудитория. Типичная поездка представителя целевой аудитории.

2. Кто эти клиенты, которые используют самокаты для поездок на длинные расстояния? Типичная длительная поездка.

3. Статистика по использованию самокатов ночью по дням недели.

**Все исследования должны сопровождаться визуализацией в виде графиков и диаграмм**


In [220]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import re
import os
os.chdir('/Users/andreypovaliy/Documents/Education/ITMO_DS/05_python_for_data_analysis/tasks/')

In [221]:
# ваш код здесь

df = pd.read_csv('kicksharing.csv')
df


order_rk    party_rk gender_cd  age education_level_cd  \
0       266071307   761067705         M   40                UGR   
1       355113920   614049469         F   30                GRD   
2       347424551   757583701         M   28                UGR   
3       351562959   541367366         M   24                GRD   
4       258647149   238473615         M   34                NaN   
...           ...         ...       ...  ...                ...   
396744  349109813   755778419         M   18                NaN   
396745  348829245  1146263947         M   23                NaN   
396746  273529251  1012601147         M   20                NaN   
396747  306062051   863877659         M   19                NaN   
396748  262579355   897372192         M   33                GRD   

       marital_status_cd         lvn_state_nm  minute_cost  activation_cost  \
0                    DIV        ТЮМЕНСКАЯ ОБЛ         4.99             30.0   
1                    MAR       РОСТОВСКАЯ ОБЛ         8.49             50.0   
2                    UNM     СВЕРДЛОВСКАЯ ОБЛ         5.99             30.0   
3                    UNM             Г МОСКВА         8.99             50.0   
4                    NaN             Г МОСКВА         6.99             50.0   
...                  ...                  ...          ...              ...   
396744               NaN     СВЕРДЛОВСКАЯ ОБЛ         6.49             30.0   
396745               NaN  КАЛИНИНГРАДСКАЯ ОБЛ         8.99             30.0   
396746               NaN        ТЮМЕНСКАЯ ОБЛ         8.79             50.0   
396747               NaN    Г САНКТ-ПЕТЕРБУРГ         7.49             50.0   
396748               UNM             Г МОСКВА         8.08             50.0   

        hold_amount transport_model  distance_km                created_dttm  \
0             300.0              SL     3.690479  2024-08-07 09:47:25.000000   
1             300.0              SL     1.114912  2024-10-17 14:57:20.000000   
2             300.0               E     1.515612  2024-09-19 05:31:41.000000   
3             300.0              SL     0.499411  2024-10-04 16:05:09.000000   
4             300.0              SL     2.602028  2024-07-10 06:57:40.000000   
...             ...             ...          ...                         ...   
396744        300.0               E     1.555069  2024-09-25 11:39:42.000000   
396745        300.0             Max     2.270902  2024-09-24 20:05:43.000000   
396746        300.0               E     1.411171  2024-09-03 16:07:43.000000   
396747        300.0              SL     1.002400  2024-09-12 18:28:48.000000   
396748        300.0              SL     0.914210  2024-07-25 07:44:07.000000   

                   book_start_dttm               book_end_dttm  \
0       2024-08-07 09:47:29.325252  2024-08-07 10:07:59.339524   
1       2024-10-17 14:57:24.586000  2024-10-17 15:04:19.419607   
2       2024-09-19 05:34:59.476000  2024-09-19 05:41:50.164372   
3       2024-10-04 16:05:13.162000  2024-10-04 16:07:31.724918   
4       2024-07-10 06:57:43.017125  2024-07-10 07:07:48.446462   
...                            ...                         ...   
396744  2024-09-25 11:39:46.885000  2024-09-25 11:45:41.545313   
396745  2024-09-24 20:05:46.954000  2024-09-24 20:13:58.592279   
396746  2024-09-03 16:07:45.311487  2024-09-03 16:28:07.510118   
396747  2024-09-12 18:28:50.401000  2024-09-12 18:32:22.911827   
396748  2024-07-25 07:44:10.998103  2024-07-25 07:50:54.825261   

        book_time_zone_cd       local_book_start_dttm  nominal_price_rub_amt  \
0                       5  2024-08-07 11:47:29.325252                 134.79   
1                       3  2024-10-17 14:57:24.586000                 109.43   
2                       5  2024-09-19 07:34:59.476000                  71.93   
3                       3  2024-10-04 16:05:13.162000                  76.97   
4                       3  2024-07-10 06:57:43.017125                 126.89   
...                  

In [222]:
df['book_end_dttm']= df['book_end_dttm'].astype('str').apply(lambda x: re.sub('.'+r'\d{6}', '', x)) 
df['book_end_dttm'] = pd.to_datetime(df['book_end_dttm'], format='%Y-%m-%d %H:%M:%S')

In [223]:
df['book_start_dttm']= df['book_start_dttm'].astype('str').apply(lambda x: re.sub('.'+r'\d{6}', '', x)) 
df['book_start_dttm'] = pd.to_datetime(df['book_start_dttm'], format='%Y-%m-%d %H:%M:%S')

In [224]:
df['duration'] = (df['book_end_dttm'] - df['book_start_dttm']).dt.total_seconds()
df['duration']

0         1230.0
1          415.0
2          411.0
3          138.0
4          605.0
           ...  
396744     355.0
396745     492.0
396746    1222.0
396747     212.0
396748     404.0
Name: duration, Length: 396749, dtype: float64

In [ ]:
df['day_of_week'] = df['book_start_dttm'].dt.day_name()
df['day_of_week']

0         Wednesday
1          Thursday
2          Thursday
3            Friday
4         Wednesday
            ...    
396744    Wednesday
396745      Tuesday
396746      Tuesday
396747     Thursday
396748     Thursday
Name: day_of_week, Length: 396749, dtype: object

In [249]:
df['dayness'] = df['book_start_dttm'].dt.hour
# .replace({True: 'day', False: 'night'})
df['dayness'] = ['night' if x < 8 else 'day' for x in df['dayness']]
df['dayness']

0           day
1           day
2         night
3           day
4         night
          ...  
396744      day
396745      day
396746      day
396747      day
396748    night
Name: dayness, Length: 396749, dtype: object

In [225]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 396749 entries, 0 to 396748
Data columns (total 21 columns):
 #   Column                   Non-Null Count   Dtype         
---  ------                   --------------   -----         
 0   order_rk                 396749 non-null  int64         
 1   party_rk                 396749 non-null  int64         
 2   gender_cd                393828 non-null  object        
 3   age                      396749 non-null  int64         
 4   education_level_cd       190059 non-null  object        
 5   marital_status_cd        217729 non-null  object        
 6   lvn_state_nm             362572 non-null  object        
 7   minute_cost              396749 non-null  float64       
 8   activation_cost          396749 non-null  float64       
 9   hold_amount              396749 non-null  float64       
 10  transport_model          396749 non-null  object        
 11  distance_km              396749 non-null  float64       
 12  created_dttm    

In [226]:
df.isna().sum()/len(df)*100

order_rk                    0.000000
party_rk                    0.000000
gender_cd                   0.736234
age                         0.000000
education_level_cd         52.095910
marital_status_cd          45.121727
lvn_state_nm                8.614262
minute_cost                 0.000000
activation_cost             0.000000
hold_amount                 0.000000
transport_model             0.000000
distance_km                 0.000000
created_dttm                0.000000
book_start_dttm             0.000000
book_end_dttm               0.000000
book_time_zone_cd           0.000000
local_book_start_dttm       0.000000
nominal_price_rub_amt       0.000000
loyalty_accrual_rub_amt     7.500208
loyalty_accrual_bns_amt    94.468039
duration                    0.000000
dtype: float64

In [191]:
df1 = df.pivot_table(values= 'order_rk', index= 'party_rk', aggfunc= "count").sort_values(by='order_rk', ascending= False)
df1.reset_index(drop=False, inplace=True)
df1

party_rk  order_rk
0      386134592       388
1      910211081       365
2      636557947       335
3      759042976       320
4      391309281       318
...          ...       ...
64047  611635328         1
64048  611533746         1
64049  611532397         1
64050  611527588         1
64051  627066884         1

[64052 rows x 2 columns]

In [192]:
df1 = df1[df1['order_rk']> 150]

In [193]:
df_subset = df[df['party_rk'].isin(df1['party_rk'])]
print(f" {np.mean(df_subset['age'])} +/- {np.std(df_subset['age'])}")



 32.775252525252526 +/- 10.934410930865765


In [194]:
df_subset['gender_cd'].value_counts()/len(df_subset)*100

gender_cd
M    85.962567
F    14.037433
Name: count, dtype: float64

In [195]:
df_subset['education_level_cd'].value_counts()/len(df_subset)*100

education_level_cd
GRD    17.595068
SCH    16.065062
UGR     7.263815
Name: count, dtype: float64

In [196]:
df_subset['marital_status_cd'].value_counts()/len(df_subset)*100

marital_status_cd
UNM    31.068033
MAR    13.740345
DIV     5.466429
CIV     2.406417
WID     1.307190
Name: count, dtype: float64

In [197]:
df_subset['lvn_state_nm'].value_counts()/len(df_subset)*100

lvn_state_nm
Г МОСКВА                   26.604278
СВЕРДЛОВСКАЯ ОБЛ           13.420974
МОСКОВСКАЯ ОБЛ             12.522282
Г САНКТ-ПЕТЕРБУРГ           9.395425
КРАСНОДАРСКИЙ КРАЙ          5.837790
РЕСП БАШКОРТОСТАН           4.396910
ТОМСКАЯ ОБЛ                 3.230838
РОСТОВСКАЯ ОБЛ              2.889186
РЕСП УДМУРТСКАЯ             2.376708
САРАТОВСКАЯ ОБЛ             2.376708
РЕСП МАРИЙ ЭЛ               1.492870
РЕСП ЛУГАНСКАЯ НАРОДНАЯ     1.336898
АСТРАХАНСКАЯ ОБЛ            1.336898
ИРКУТСКАЯ ОБЛ               1.292335
САНКТ-ПЕТЕРБУРГ             1.262626
УЛЬЯНОВСКАЯ ОБЛ             1.225490
ОРЛОВСКАЯ ОБЛ               1.158645
ОБЛ СВЕРДЛОВСКАЯ            1.151218
РЕСП ТАТАРСТАН              1.136364
Name: count, dtype: float64

In [198]:
np.mean(df_subset['duration'])/60

np.float64(10.055108684888097)

In [199]:
np.mean(df_subset['distance_km'])

np.float64(3.6407163454346474)

In [200]:
np.mean(df_subset['nominal_price_rub_amt'])

np.float64(115.07178698752229)

In [201]:
df_subset['transport_model'].value_counts()/len(df_subset)*100

transport_model
SL              57.189542
E               24.970291
Max             14.230541
ES400            2.911468
Urban E-Bike     0.646168
ES100            0.029709
Techno           0.022282
Name: count, dtype: float64

**1. Кто эти клиенты, которее чаще других использует сервис Самокаты? Целевая аудитория.** 

Холостой мужчина с высшым или начальным образованием со средним возрастом 32.77 +/- 10.93 лет, проживающий перимуществено в Москве.


**Типичная поездка представителя целевой аудитории.**

Средняя поездка в пределах 10 минут на расстояние 3.64 км со стоимостью 115.07 рублей преимущественно на самокате SL.

In [230]:
df2 = df.sort_values(by = 'duration', ascending= False)
df2 = df[df['duration'] > 10000.0]
df2
# df.describe()

order_rk    party_rk gender_cd  age education_level_cd  \
9189    273798957   157834082         M   37                GRD   
10286   254952282   647055790         M   40                SCH   
10419   332643712  1078173708         M   23                NaN   
12188   347724287   262948834         M   43                NaN   
12350   277398952   850581040         F   71                UGR   
...           ...         ...       ...  ...                ...   
389957  268543359   341502281         M   48                SCH   
392866  347143130  1162431699         M   34                NaN   
393095  269077423     4799493         F   35                GRD   
393624  268542805   529050807         M   20                NaN   
396437  350276685   430254020         M   25                NaN   

       marital_status_cd         lvn_state_nm  minute_cost  activation_cost  \
9189                 MAR       ТАМБОВСКАЯ ОБЛ         6.49             40.0   
10286                UNM       ТАМБОВСКАЯ ОБЛ         7.49             20.0   
10419                NaN                  NaN         8.31             50.0   
12188                NaN        ТЮМЕНСКАЯ ОБЛ         5.99             30.0   
12350                MAR        САМАРСКАЯ ОБЛ         8.29             50.0   
...                  ...                  ...          ...              ...   
389957               MAR  КАЛИНИНГРАДСКАЯ ОБЛ         6.99             50.0   
392866               NaN       МОСКОВСКАЯ ОБЛ         5.99             40.0   
393095               MAR             Г МОСКВА         7.49             50.0   
393624               UNM    КРАСНОЯРСКИЙ КРАЙ         9.29             50.0   
396437               NaN        САМАРСКАЯ ОБЛ         8.99             50.0   

        hold_amount  ... distance_km                created_dttm  \
9189          300.0  ...   10.757139  2024-09-04 05:42:14.000000   
10286         300.0  ...   10.743667  2024-06-26 16:47:00.000000   
10419         300.0  ...   63.815338  2024-09-15 11:34:54.000000   
12188         300.0  ...   33.369161  2024-09-20 07:46:17.000000   
12350         300.0  ...   15.976148  2024-09-10 09:53:46.000000   
...             ...  ...         ...                         ...   
389957        300.0  ...    2.933850  2024-08-16 08:18:43.000000   
392866        300.0  ...   18.861971  2024-09-18 10:41:41.000000   
393095        300.0  ...   36.095742  2024-08-18 10:53:50.000000   
393624        300.0  ...   14.402980  2024-08-16 15:52:07.000000   
396437        300.0  ...   22.558165  2024-09-29 14:00:58.000000   

           book_start_dttm       book_end_dttm book_time_zone_cd  \
9189   2024-09-04 05:42:18 2024-09-04 08:47:14                 3   
10286  2024-06-26 16:47:03 2024-06-26 20:57:16                 3   
10419  2024-09-15 11:34:58 2024-09-15 15:46:54                 3   
12188  2024-09-20 07:46:21 2024-09-20 10:41:12                 5   
12350  2024-09-10 09:53:57 2024-09-10 12:55:10                 3   
...                    ...                 ...               ...   
389957 2024-08-16 08:18:48 2024-08-16 11:21:07                 2   
392866 2024-09-18 10:41:45 2024-09-18 13:53:50                 3   
393095 2024-08-18 10:53:53 2024-08-18 13:48:35                 3   
393624 2024-08-16 15:52:15 2024-08-16 18:43:56                 3   
396437 2024-09-29 14:01:00 2024-09-29 17:23:58                 3   

             local_book_start_dttm nominal_price_rub_amt  \
9189    2024-09-04 05:42:18.815635               1240.65   
10286   2024-06-26 16:47:03.650487               1862.54   
10419   2024-09-15 11:34:58.482000               2144.12   
12188   2024-09-20 09:46:21.645000               1078.25   
12350   2024-09-10 09:53:57.191000               1558.78   
...                            ...                   ...   
389957  2024-08-16 07:18:48.095850               1329.17   
392866  2024-09-18 10:41:45.916000               1196.07   
393095  2024-08-18 10:53:53.696638               1360.75   
393624  2024-08-16 15

In [231]:
print(f" {np.mean(df2['age'])} +/- {np.std(df2['age'])}")

 34.35265700483092 +/- 11.99421278297665


In [232]:
df2['gender_cd'].value_counts()/len(df2)*100

gender_cd
M    71.014493
F    28.502415
Name: count, dtype: float64

In [233]:
df2['lvn_state_nm'].value_counts()/len(df2)*100

lvn_state_nm
Г МОСКВА                                       17.391304
Г САНКТ-ПЕТЕРБУРГ                               6.280193
МОСКОВСКАЯ ОБЛ                                  6.280193
РЕСП ТАТАРСТАН                                  5.314010
НОВОСИБИРСКАЯ ОБЛ                               3.381643
КРАСНОЯРСКИЙ КРАЙ                               3.381643
КРАСНОДАРСКИЙ КРАЙ                              3.381643
ТЮМЕНСКАЯ ОБЛ                                   3.381643
СВЕРДЛОВСКАЯ ОБЛ                                3.381643
САМАРСКАЯ ОБЛ                                   2.898551
НИЖЕГОРОДСКАЯ ОБЛ                               2.898551
ЛЕНИНГРАДСКАЯ ОБЛ                               2.415459
ЧЕЛЯБИНСКАЯ ОБЛ                                 2.415459
ТАМБОВСКАЯ ОБЛ                                  1.932367
УЛЬЯНОВСКАЯ ОБЛ                                 1.449275
ИРКУТСКАЯ ОБЛ                                   1.449275
ВОЛГОГРАДСКАЯ ОБЛ                               1.449275
КЕМЕРОВСКАЯ ОБЛАСТ

In [234]:
df2['marital_status_cd'].value_counts()/len(df2)*100

marital_status_cd
UNM    29.468599
MAR    19.323671
CIV     3.864734
DIV     2.898551
WID     0.483092
Name: count, dtype: float64

In [235]:
df2['education_level_cd'].value_counts()/len(df2)*100

education_level_cd
GRD    24.154589
SCH    15.458937
UGR     8.695652
PGR     2.898551
Name: count, dtype: float64

In [236]:
np.mean(df2['duration'])/60

np.float64(228.6563607085346)

In [238]:
np.mean(df2['distance_km'])

np.float64(60.93362591967729)

In [239]:
np.mean(df2['nominal_price_rub_amt'])

np.float64(1751.19729468599)

In [240]:
df2['transport_model'].value_counts()/len(df2)*100

transport_model
SL              62.801932
Max             20.289855
E               10.628019
Urban E-Bike     2.898551
ES200            0.966184
ES400            0.966184
Lucky            0.483092
LB               0.483092
ES100            0.483092
Name: count, dtype: float64

**2. Кто эти клиенты, которые используют самокаты для поездок на длинные расстояния?** 

Холостой мужчина с высшым образованием со средним возрастом 34.49 +/- 9.09 лет, проживающий перимуществено в Москве.

**Типичная поездка.**

Средняя поездка в пределах 229 минут на расстояние 60.93 км со стоимостью 1751.19 рублей преимущественно на самокате SL.

**3. Статистика по использованию самокатов ночью по дням недели.**

In [259]:
df3 = df[df['dayness'] == 'night']
df3


order_rk    party_rk gender_cd  age education_level_cd  \
2       347424551   757583701         M   28                UGR   
4       258647149   238473615         M   34                NaN   
7       265062394   866880584         M   38                NaN   
12      349990173   676025738         F   34                GRD   
15      267646217   397325703         M   32                NaN   
...           ...         ...       ...  ...                ...   
396725  267392548  1067967329         M   20                NaN   
396726  306066373   145120243         M   42                GRD   
396728  270437089   326013844         M   29                GRD   
396736  354838600   108170194         M   42                PGR   
396748  262579355   897372192         M   33                GRD   

       marital_status_cd        lvn_state_nm  minute_cost  activation_cost  \
2                    UNM    СВЕРДЛОВСКАЯ ОБЛ         5.99             30.0   
4                    NaN            Г МОСКВА         6.99             50.0   
7                    NaN    СВЕРДЛОВСКАЯ ОБЛ         5.99             30.0   
12                   UNM    СВЕРДЛОВСКАЯ ОБЛ         5.99             30.0   
15                   UNM    СВЕРДЛОВСКАЯ ОБЛ         5.49             30.0   
...                  ...                 ...          ...              ...   
396725               NaN        ТВЕРСКАЯ ОБЛ         6.99             30.0   
396726               MAR            Г МОСКВА         6.53             50.0   
396728               MAR  КРАСНОДАРСКИЙ КРАЙ         4.99             35.0   
396736               MAR            Г МОСКВА         7.49             50.0   
396748               UNM            Г МОСКВА         8.08             50.0   

        hold_amount  ...     book_start_dttm       book_end_dttm  \
2             300.0  ... 2024-09-19 05:34:59 2024-09-19 05:41:50   
4             300.0  ... 2024-07-10 06:57:43 2024-07-10 07:07:48   
7             300.0  ... 2024-08-03 03:09:38 2024-08-03 03:18:47   
12            300.0  ... 2024-09-28 06:35:08 2024-09-28 06:53:57   
15            300.0  ... 2024-08-13 03:38:28 2024-08-13 03:40:25   
...             ...  ...                 ...                 ...   
396725        300.0  ... 2024-08-12 07:14:48 2024-08-12 07:35:14   
396726        300.0  ... 2024-09-12 06:09:29 2024-09-12 06:12:00   
396728        300.0  ... 2024-08-23 04:25:57 2024-08-23 04:27:44   
396736        300.0  ... 2024-10-16 06:39:45 2024-10-16 06:47:54   
396748        300.0  ... 2024-07-25 07:44:10 2024-07-25 07:50:54   

       book_time_zone_cd       local_book_start_dttm nominal_price_rub_amt  \
2                      5  2024-09-19 07:34:59.476000                 71.93   
4                      3  2024-07-10 06:57:43.017125                126.89   
7                      5  2024-08-03 05:09:38.493851                 89.90   
12                     5  2024-09-28 08:35:08.384000                143.81   
15                     5  2024-08-13 05:38:28.873442                 40.98   
...                  ...                         ...                   ...   
396725                 3  2024-08-12 07:14:48.753791                176.79   
396726                 3  2024-09-12 06:09:29.282000                 69.59   
396728                 3  2024-08-23 04:25:57.786804                 44.98   
396736                 3  2024-10-16 06:39:45.500000                117.41   
396748                 3  2024-07-25 07:44:10.998103                106.56   

        loyalty_accrual_rub_amt loyalty_accrual_bns_amt  duration  \
2                         10.79                     NaN     411.0   
4                         25.38                     NaN     605.0   
7                         12.99                     NaN     549.0   
12                        63.76                     NaN    1129.0   
15                        12.29                     NaN     117.0   
...                         ...                     ...       ...   
396725                    61.0

In [262]:
df3.groupby("day_of_week")[["duration"]].agg(["describe"])

duration                                                     \
            describe                                                      
               count        mean         std   min    25%    50%    75%   
day_of_week                                                               
Friday       15923.0  550.388369  596.565566  13.0  267.0  405.0  638.0   
Monday       15135.0  538.825768  533.272035  11.0  258.0  398.0  634.0   
Saturday     10301.0  645.812154  799.878355  11.0  276.0  444.0  748.0   
Sunday        8681.0  670.653151  813.498520  10.0  275.0  450.0  758.0   
Thursday     15956.0  542.723991  543.748935   9.0  262.0  403.0  635.0   
Tuesday      15399.0  547.948438  539.540091  11.0  264.0  407.0  645.0   
Wednesday    15849.0  540.184113  556.015167  11.0  260.0  400.0  626.0   

                      
                      
                 max  
day_of_week           
Friday       29812.0  
Monday       13773.0  
Saturday     37017.0  
Sunday       17381.0  
Thursday     13122.0  
Tuesday      11833.0  
Wednesday    11702.0

In [263]:
df3.groupby("day_of_week")["nominal_price_rub_amt"].agg(["describe"])

describe                                                     \
               count        mean        std  min     25%    50%     75%   
day_of_week                                                               
Friday       15923.0  105.980753  68.550778  0.0  75.960  94.91  119.90   
Monday       15135.0  104.464734  63.325382  0.0  75.125  92.91  119.56   
Saturday     10301.0  119.971985  96.433796  0.0  77.960  98.93  132.39   
Sunday        8681.0  123.030775  97.121462  0.0  79.900  99.74  133.88   
Thursday     15956.0  104.797649  64.024176  0.0  75.430  94.12  118.86   
Tuesday      15399.0  105.499816  63.425432  0.0  75.960  94.81  119.85   
Wednesday    15849.0  105.181049  65.901478  0.0  75.960  94.87  118.48   

                      
                 max  
day_of_week           
Friday       3007.03  
Monday       1982.70  
Saturday     4362.83  
Sunday       2222.10  
Thursday     1747.09  
Tuesday      1640.99  
Wednesday    1655.24

In [264]:
df3.groupby("day_of_week")["age"].agg(["describe"])

describe                                                   
               count       mean       std   min   25%   50%   75%   max
day_of_week                                                            
Friday       15923.0  31.656660  9.059851  14.0  24.0  31.0  37.0  78.0
Monday       15135.0  31.530096  9.246925  18.0  24.0  31.0  37.0  87.0
Saturday     10301.0  31.291816  9.423194  18.0  24.0  30.0  37.0  80.0
Sunday        8681.0  31.660638  9.613967  18.0  24.0  31.0  37.0  84.0
Thursday     15956.0  31.738844  9.100404  16.0  24.0  31.0  37.0  85.0
Tuesday      15399.0  31.758426  9.191739  17.0  24.0  31.0  38.0  84.0
Wednesday    15849.0  31.665973  9.104696  14.0  24.0  31.0  37.0  84.0